File: src/main/kotlin/dev/alexnader/framity/gui/SingleItemSlot.kt
package dev.alexnader.framity.gui

import io.github.cottonmc.cotton.gui.ValidatedSlot
import io.github.cottonmc.cotton.gui.widget.WItemSlot
import net.minecraft.inventory.Inventory
import net.minecraft.item.ItemStack

open class SingleItemSlot(
    inventory: Inventory,
    startIndex: Int,
    slotsWide: Int = 1,
    slotsHigh: Int = 1,
    big: Boolean = false
) : WItemSlot(inventory, startIndex, slotsWide, slotsHigh, big) {
    override fun createSlotPeer(inventory: Inventory, index: Int, x: Int, y: Int) =
        SingleSlot(inventory, index, x, y)
}

class SingleSlot(inventory: Inventory, index: Int, x: Int, y: Int) : ValidatedSlot(inventory, index, x, y) {
    override fun getMaxStackAmount() =
        1

    override fun getMaxStackAmount(stack: ItemStack?) =
        1
}

File: src/main/scala/dev/alexnader/framity/gui/SingleItemSlot.scala
package dev.alexnader.framity.gui

import io.github.cottonmc.cotton.gui.ValidatedSlot
import net.minecraft.inventory.Inventory
import net.minecraft.item.ItemStack

object SingleItemSlot extends ((Inventory, Int, Int, Int) => SingleItemSlot) {
  override def apply(v1: Inventory, v2: Int, v3: Int, v4: Int): SingleItemSlot = new SingleItemSlot(v1, v2, v3, v4)
}

class SingleItemSlot(inventory: Inventory, index: Int, x: Int, y: Int) extends ValidatedSlot(inventory, index, x, y) {
  override def getMaxStackAmount: Int = 1

  override def getMaxStackAmount(stack: ItemStack): Int = 1
}
